### PR TITLE
fix: add init.py to the annotations module

### PR DESCRIPTION
**Pull request by bkyryliuk**
### SUMMARY
Small fixes for the https://github.com/apache/incubator-superset/pull/11344

Adds init file and fixes mypy issues

### TEST PLAN
Unit tests

### CODE REVIEW

File: superset/annotation_layers/annotations/commands/update.py [hunk 1]
```diff
@@ -70,15 +70,15 @@ def validate(self) -> None:
             if not annotation_layer:
                 raise AnnotationLayerNotFoundError()
             self._properties["layer"] = annotation_layer
+
+            # Validate short descr uniqueness on this layer
+            if not AnnotationDAO.validate_update_uniqueness(
+                layer_id, short_descr, annotation_id=self._model_id,
```
Comment (bkyryliuk) [inline superset/annotation_layers/annotations/commands/update.py:76]: layer_id is not optional -> moving it under if layer_id

**Comment by bkyryliuk** [comment]
eh, looks like more fixes needed like linter etc :(

**Review by dpgaspar — APPROVED** [review]
Looks good, thks (just some lint issues on CI)

File: superset/annotation_layers/annotations/api.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ class AnnotationRestApi(BaseSupersetModelRestApi):
     openapi_spec_methods = openapi_spec_methods_override
 
     @staticmethod
-    def _apply_layered_relation_to_rison(layer_id: int, rison_parameters) -> None:
+    def _apply_layered_relation_to_rison(layer_id: int, rison_parameters: Any) -> None:
```
Comment (villebro) [inline superset/annotation_layers/annotations/api.py:119]: Nit: I think it's
```suggestion
    def _apply_layered_relation_to_rison(layer_id: int, rison_parameters: Dict[str, Any]) -> None:
```

**Review by villebro — APPROVED** [review]
LGTM, one small typing nit

**Comment by rusackas** [comment]
A rebase _should_ fix the broken E2E test (see #11453)

**Pull request merged**